pkg/scale/decode.go

```diff
@@ -105,15 +113,29 @@ func (d *Decoder) Decode(dst interface{}) (err error) {
 }
 
 // NewDecoder is constructor for Decoder
-func NewDecoder(r io.Reader) (d *Decoder) {
-	d = &Decoder{
-		decodeState{r},
+func NewDecoder(r io.Reader, options ...func(*Decoder)) (d *Decoder) {
```

Comment:
What is the `option` here for exactly?
Are they like helper functions for decoding? In which case, may be they could be passed only here and be saved as property of `decoder` and use them in the `Decode` function. If that makes sense, decode function would not need to be passed options.

Reply:
That makes sense. Although you would still have to pass options to `scale.Unmarshal` which creates its decoder within AFAIK.